### DISCUSSION TITLE
Using Verdaccio with Nugat #C

### DISCUSSION THREAD

**Discussion by bolti95** [verdaccio/verdaccio#3179] [Q&A]
I'd like to create a private registry for C# packages using Nugat and verdaccio, Nugat's package API (https://api.nuget.org/v3/index.json) requires a token, I want to know, how can I set up Verdaccio with this registry?

Following the example of how to publish to [private github repository](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry#authenticating-with-a-personal-access-token), it might be some sort of configuration like this:
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
    <packageSources>
        <clear />
        <add key="github" value="https://nuget.pkg.github.com/OWNER/index.json" />
    </packageSources>
    <packageSourceCredentials>
        <github>
            <add key="Username" value="USERNAME" />
            <add key="ClearTextPassword" value="TOKEN" />
        </github>
    </packageSourceCredentials>
</configuration>
```

**Comment by juanpicado** [comment]
hi @bolti95 

I have no much idea how Nugat works or is capable to, but, does it works with npmjs registry ? I know GitHub registry support https://docs.github.com/en/packages/learn-github-packages/introduction-to-github-packages#support-for-package-registries with Nugat but I guess is not the case for verdaccio. Maybe you can elaborate a bit more if I'm missing something here.

**Reply by bolti95** [reply]
I was wondering if I could set this as a registry like I would for NPM :) but it uses an API in order to access

**Comment by bolti95** [comment]
@juanpicado I was able to upload NuGet packages to NPM registry, for Unity purposes, by creating a dll file, as explained [in this tutorial](https://www.youtube.com/watch?v=GRn49ehm_pI), and a package.json

**Reply by bolti95** [reply]
I had to have NuGet for Unity installed as a package via the registry in order for this to work
